src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -219,6 +235,28 @@ private ShiftDays generateShiftDays(Set<Integer> shiftDays) throws IllegalValueE
         return new ShiftDays(shiftDays);
     }
 
+    /**
+     * Generates a set of {@link Specialisation} object from the stored JSON data (only for Doctors).
+     *
+     * @param specialisations The person's specialisations.
+     * @return A set of {@link Specialisation} object based on the stored JSON data.
+     * @throws IllegalValueException If the role is not "Doctor" or specialisations are null or do not meet the
+     *      constraints.
+     */
+    private Set<Specialisation> generateSpecialisations(Set<String> specialisations) throws IllegalValueException {
+        if (!role.equals("Doctor") && specialisations == null) {
```

Comment:
I think this should check if the role is Doctor instead. That way in the JSON data file, if the particular Person is indicated as a Doctor and it does not have the specialisation attribute, the exception can be thrown.

Reply:
Yup, make sense